main.tf

```diff
@@ -1041,6 +1041,77 @@ resource "aws_nat_gateway" "this" {
   depends_on = [aws_internet_gateway.this]
 }
 
+resource "aws_route" "private_route_custom" {
+  # NOTE(bogdando): serialize rules as JSON for setproduct() to match
+  # its string data type of route tables IDs, then deserialize from json to
+  # consume it as end values. Iterate over a map with keys names matching
+  # router IDs (key unquieness is ensured by a hash of each rule)
+  # as for_each cannot take these from a list.
+  for_each = {
+    for elements in chunklist(flatten([
+      for pair in setproduct(
+        aws_route_table.private[*].id,
+        [var.private_routes_extra]
+      ) : setproduct([pair[0]], [for p in pair[1] : [jsonencode(p)]])
+    ]), 2) : "${elements[0]}:${sha256(elements[1])}" => jsondecode(elements[1])
```

Comment:
Sorry, but this code is not what we want to accept in this module...

Reply:
Thank you for feedback. I couldn't find another approach to make setproduct and for_each working on different data types - strings for route tables IDs and lists of maps for roles. If you have ideas to share, I'd appreciate. I'm quite new to terraform, so please excuse my ignorance.

Reply:
Please also note linked terragrunt issues,. It seems there is no github URL to use aws_route as a dedicated module alongside vpc one